Application/Frontend/pages/api/registration.ts

```diff
@@ -39,12 +49,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         email: email,
         phone: phone,
         createdAt: createdAt,
-        friendsList: ["user1", "user2"] // for now every single new user is friends with user1 and user2. This would be replaced with an add friends button.
-        }); // IMPORTANT: The findOne method returns a promise, so we need to await the resolution of the promise first
-      return res.status(200).json({ message: 'Registration success' });
+        friendsList: ["user1", "user2"] // Example initial friends list
+      });
 
+      return res.status(200).json({ message: 'Registration successful' });
     } catch (error) {
-      console.error(error);
+      console.error('Failed to register user:', error);
```

Comment:
- Enhance error handling to provide more informative error messages to clients and aid in troubleshooting. Instead of a generic "Internal server error" message, consider providing more context about the specific error that occurred.